core/net/rpl/rpl-ext-header.c

```diff
@@ -179,9 +179,11 @@ rpl_srh_get_next_hop(uip_ipaddr_t *ipaddr)
 {
   uint8_t *uip_next_hdr;
   int last_uip_ext_len = uip_ext_len;
+  rpl_dag_t *dag;
 
   uip_ext_len = 0;
   uip_next_hdr = &UIP_IP_BUF->proto;
+  dag = rpl_get_dag(&UIP_IP_BUF->destipaddr);
```

Comment:
Seems wrong to me. The IPv6 destination here is the next hop in SRH. Why would there be root of a DAG?